content/blog/censored-0-2-0/index.Rmd

```diff
@@ -0,0 +1,158 @@
+The documentation for the [aorsf](https://docs.ropensci.org/aorsf) package includes a nice illustration of this with the splits for an axis-based tree on the left and an oblique tree on the right:
+
+![Two scatter plots of data with two predictors, X1 and X2, and two classes, coded as pink dots and orange squares. The lefthand plot shows the splits of an axis-based decision tree which are at a right angle to the axis. The resulting partition generally separates the classes well but not perfectly. The righthand plot shows the splits of an oblique tree which achieves perfect separation on this example because it can cut across the predictor space diagnonally.](https://docs.ropensci.org/aorsf/reference/figures/tree_axis_v_oblique.png)
+
+To fit such a model, set the engine for a random forest to `"aorsf"`:
+
+```{r}
+lung <- na.omit(lung)
+
+forest <- rand_forest() |> 
+  set_engine("aorsf") |> 
+  set_mode("censored regression") |> 
+  fit(Surv(time, status) ~ ., data = lung)
+
+pred <- predict(forest, lung[1:3, ], type = "survival", eval_time = c(100, 500))
+pred$.pred[[1]]
+```
+
+### New `"flexsurvspline"` engine for `survival_reg()`
+
+This engine has been contributed by [Matt Warkentin](https://github.com/mattwarkentin) and enables users to fit a parametric survival model with splines via `flexsurv::flexsurvspline()`. 
+
+This model uses natural cubic splines to model a transformation of the survival function, e.g., the log cumulative hazard. This gives a lot more flexibility to a parametric model allowing us, for example, to represent more irregular hazard curves. Let's illustrate that with a data set of survival times of breast cancer patients, based on the example from [Jackson (2016)](https://www.jstatsoft.org/article/view/v070i08). 
+
+The flexibility of the model is governed by `k`, the number of knots in the spline. We set `scale = "odds"` for a proportional hazards model. 
+
+```{r}
+data(bc, package = "flexsurv")
+
+fit_splines <- survival_reg() |> 
+  set_engine("flexsurvspline", k = 5, scale = "odds") |> 
+  fit(Surv(recyrs, censrec) ~ group, data = bc)
+```
+
+For comparison, we also fit a parametric model without splines.
+
+```{r}
+fit_gengamma <- survival_reg(dist = "gengamma") |> 
+  set_engine("flexsurv") |> 
+  fit(Surv(recyrs, censrec) ~ group, data = bc)
+```
+
+We can predict the hazard for the three levels of the prognostic `group`
+
+```{r}
+bc_groups <- tibble(group = c("Poor","Medium","Good"))
+
+pred_splines <- predict(fit_splines, new_data = bc_groups, type = "hazard", 
+                        eval_time = seq(0.1, 8, by = 0.1)) |> 
+  mutate(model = "splines") |> 
+  bind_cols(bc_groups)
+
+pred_gengamma <- predict(fit_gengamma, new_data = bc_groups, type = "hazard", 
+                         eval_time = seq(0.1, 8, by = 0.1)) |> 
+  mutate(model = "gengamma") |> 
+  bind_cols(bc_groups)
+```
+
+and plot the predictions of both models which show a lot more flexibility for the splines model.
+
+```{r, fig.alt = ""}
```

Comment:
Could we add alt text here? Or did you feel that it'd be redundant with the prepended description and printed code?

Reply:
The answer is yes :D I did initially think it would read as redundant and I think I recall the advice to set the alt text to blank in such situations (rather than not adding any). But I've now added non-redundant alt text, that's better. Thanks for the nudge!